### ISSUE TITLE
Critical dependency

### ISSUE THREAD

**Issue by nudl3s** [davidstutz/bootstrap-multiselect#1137]
Hi guys, 

I've run 'yarn add bootstrap-multiselect'
In my symfony 5 project in app.js (second screenshot) - require("bootstrap-multiselect");
and after running yarn build I am receiving a warning and still not working:

warning  in ./node_modules/bootstrap-multiselect/dist/js/bootstrap-multiselect.js
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

![Screenshot_11](https://user-images.githubusercontent.com/17205764/79066903-f63ed000-7cc3-11ea-9dbc-bd2ba343d8f8.png)

![Screenshot_10](https://user-images.githubusercontent.com/17205764/79066891-e1623c80-7cc3-11ea-8cd7-4bf4c0a3aff1.png)

**Comment by chilio** [comment]
HI @nudl3s,
I have the same problem. 
Did you find a solution?

**Comment by nudl3s** [comment]
still not mate, maybe its not made for webpack ...

**Comment by FredDut** [comment]
Hi.
I've read somewhere (I don't remember where) you need to comment few lines in dist/js/bootstrap-multiselect.js, like this 
`//if (typeof define === 'function' && define.amd &&
     //   typeof require === 'function' && typeof require.specified === 'function' && require.specified('knockout')) {

        // AMD. Register as an anonymous module.
        define(['jquery', 'knockout'], factory);
   // } else {
        // Browser globals
    //    factory(root.jQuery, root.ko);
    //}
`
It works for me.

**Comment by TheNephalim** [comment]
> 

The issue is mostly annoying because webpack will still create the bundle.

I tried the fix suggested by @FredDut and it worked.  

However, I also ran across this:  [https://gitter.im/webpack/webpack?at=57ad346cae838f6f56991e5a](url).  

Based on the suggestion on Gitter, I added the following to the webpack config file:

        plugins: [
            new webpack.DefinePlugin({
                "require.specified": "require.resolve"
            }),

I took a look at the resulting bundle and it looks like webpack replaces require.specified with the following:

    // check to see if 'knockout' AMD module is specified if using requirejs
    if (typeof define === 'function' && define.amd &&
        "function" === 'function' && "function" === 'function' && /*require.resolve*/(/*! knockout */ "./node_modules/knockout/build/output/knockout-latest.js")) {

        // AMD. Register as an anonymous module.
        define(['jquery', 'knockout'], factory);
    } else {
        // Browser globals
        factory(__webpack_provided_window_dot_jQuery, root.ko);
    }

Bottom line is that webpack didn't complain.  

I hope this helps someone.
